fili-core/src/main/java/com/yahoo/bard/webservice/data/config/table/ConcretePhysicalTableDefinition.java

```diff
@@ -34,7 +36,30 @@ public ConcretePhysicalTableDefinition(
             Set<FieldName> metricNames,
             Set<? extends DimensionConfig> dimensionConfigs
     ) {
-        super(name, timeGrain, metricNames, dimensionConfigs);
+        super(name, timeGrain, metricNames, dimensionConfigs, null, null);
+    }
+
+    /**
+     * Define a physical table using a zoned time grain.
+     *
+     * @param name  The table name
+     * @param timeGrain  The zoned time grain
+     * @param metricNames  The Set of metric names on the table
+     * @param dimensionConfigs  The dimension configurations
+     * @param expectedStartDate  The expected start date of the datasource the constructed table will represent. Null
+     * indicates there is NO expected start date
+     * @param expectedEndDate  The expected end date of the datasource the constructed table will represent. Null
+     * indicates there is NO expected end date
+     */
+    public ConcretePhysicalTableDefinition(
+            TableName name,
+            ZonedTimeGrain timeGrain,
+            Set<FieldName> metricNames,
+            Set<? extends DimensionConfig> dimensionConfigs,
+            DateTime expectedStartDate,
+            DateTime expectedEndDate
```

Comment:
Looks like the two dates are optional. Since constructor arguments are pilling up with optional arguments, shall we make use builder pattern for  `ConcretePhysicalTableDefinition` and `PhysicalTableDefinition`?

Reply:
We can't eliminate the old constructors until we move to 0.12

As for not being optional, my reasoning is that these are config objects, so it seems reasonable to maintain the nulls and let the actual model object decide what to do with the nulls. In this case it is wrap them in optionals, but still.

Reply:
wait for 0.12 because we can't break public method and constructor contracts without a version bump. Just something that was agreed to after similar changes caused a lot of pain for downstream projects.

Reply:
Adding a builder pattern makes sense, feel free to spawn an issue.  We should look at how comprehensively we want to do that.
Obviously not in this PR.